Orbitersdk/samples/ProjectApollo/src_launch/mcc.h

```diff
@@ -295,7 +296,7 @@ class MCC {
 	// SUBTHREAD MANAGEMENT
 	int subThreadMode;										// What should the subthread do?
 	int subThreadType;										// What type of subthread?
-	int subThreadStatus;									// 0 = done/not busy, 1 = busy, negative = done with error
+	std::atomic<int> subThreadStatus;						// 0 = done/not busy, 1 = busy, negative = done with error
```

Comment:
I can do that but I'd recommend doing the same in ARCore to keep things aligned.
Also I have not seen a single instance of subThreadStatus being set to a negative value so the comment is likely wrong ; however there is a '2' value to indicate it is running ('1' is kind of a 'scheduled but not yet running' state).
Do we want to have the option to specify different error codes or just a single value indicating an unspecified error? (it would be easier to use an enum class this way).
I would go with this :
```
enum class ThreadStatus {
   Error,
   Stopped,
   Scheduled,
   Running
};
```
What do you think?

Reply:
Pushed a modification for this. An enum is used instead of an int, the change has been done for in the ARCore too to keep things the same.